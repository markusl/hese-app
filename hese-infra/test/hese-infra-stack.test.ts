import * as cdk from 'aws-cdk-lib';
import { SynthUtils } from '@aws-cdk/assert';
import { HeseInfraStack } from '../lib/hese-infra-stack';
import { snapshotSerializer } from './util';

expect.addSnapshotSerializer(snapshotSerializer);

test('HeseInfraStack synthetizes expected template', async () => {
  const app = new cdk.App();
  const stack = new HeseInfraStack(app, 'infra');

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
