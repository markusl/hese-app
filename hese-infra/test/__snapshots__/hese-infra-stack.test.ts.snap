// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeseInfraStack synthetizes expected template 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "HeseInfraPipeline2802B9AF": {
      "DependsOn": [
        "HeseInfraPipelineRoleDefaultPolicy9CD0A2E3",
        "HeseInfraPipelineRole42FEB153",
      ],
      "Properties": {
        "ArtifactStore": {
          "Location": {
            "Ref": "HeseInfraPipelineArtifactsBucket79D993E5",
          },
          "Type": "S3",
        },
        "RestartExecutionOnUpdate": true,
        "RoleArn": {
          "Fn::GetAtt": [
            "HeseInfraPipelineRole42FEB153",
            "Arn",
          ],
        },
        "Stages": [
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "AWS",
                  "Provider": "CodeStarSourceConnection",
                  "Version": "1",
                },
                "Configuration": {
                  "BranchName": "master",
                  "ConnectionArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:codestar-connections:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":connection/2176abff-fac4-4c5d-87e8-0cc53551ab98",
                      ],
                    ],
                  },
                  "FullRepositoryId": "markusl/hese-app",
                },
                "Name": "markusl_hese-app",
                "OutputArtifacts": [
                  {
                    "Name": "markusl_hese_app_Source",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineSourcemarkuslheseappCodePipelineActionRole27EDF73B",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "EnvironmentVariables": "[{"name":"_PROJECT_CONFIG_HASH","type":"PLAINTEXT","value":"d97537e5d14964cfc6d6890bb95a70c3d967e7019b0eb90e5c6f066668368363"}]",
                  "ProjectName": {
                    "Ref": "HeseInfraPipelineBuildSynthCdkBuildProjectAD5CA4AD",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "markusl_hese_app_Source",
                  },
                ],
                "Name": "Synth",
                "OutputArtifacts": [
                  {
                    "Name": "Synth_Output",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineCodeBuildActionRole5463EFD7",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "EnvironmentVariables": "[{"name":"_PROJECT_CONFIG_HASH","type":"PLAINTEXT","value":"7cf8d089fc7cd92c9bf13bafeb66273ae6c7f631d474a7bea6d9ac2ddd0cfff1"}]",
                  "ProjectName": {
                    "Ref": "HeseInfraPipelineUpdatePipelineSelfMutationD83C5804",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "Synth_Output",
                  },
                ],
                "Name": "SelfMutate",
                "RoleArn": {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineCodeBuildActionRole5463EFD7",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "UpdatePipeline",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "HeseInfraPipelineArtifactsBucket79D993E5": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "HeseInfraPipelineArtifactsBucketPolicyEFC8C935": {
      "Properties": {
        "Bucket": {
          "Ref": "HeseInfraPipelineArtifactsBucket79D993E5",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineArtifactsBucket79D993E5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HeseInfraPipelineArtifactsBucket79D993E5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "HeseInfraPipelineBuildSynthCdkBuildProjectAD5CA4AD": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "Description": "Pipeline step infra/Pipeline/Build/Synth",
        "EncryptionKey": "alias/aws/s3",
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:7.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "HeseInfraPipelineBuildSynthCdkBuildProjectRoleAA3DC00D",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": "{
  "version": "0.2",
  "phases": {
    "build": {
      "commands": [
        "n 20",
        "cd hese-infra",
        "npm ci",
        "npm run test",
        "npx cdk synth"
      ]
    }
  },
  "artifacts": {
    "base-directory": "hese-infra/cdk.out",
    "files": "**/*"
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "HeseInfraPipelineBuildSynthCdkBuildProjectRoleAA3DC00D": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "HeseInfraPipelineBuildSynthCdkBuildProjectRoleDefaultPolicyF237693C": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "HeseInfraPipelineBuildSynthCdkBuildProjectAD5CA4AD",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "HeseInfraPipelineBuildSynthCdkBuildProjectAD5CA4AD",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "HeseInfraPipelineBuildSynthCdkBuildProjectAD5CA4AD",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineArtifactsBucket79D993E5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HeseInfraPipelineArtifactsBucket79D993E5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HeseInfraPipelineBuildSynthCdkBuildProjectRoleDefaultPolicyF237693C",
        "Roles": [
          {
            "Ref": "HeseInfraPipelineBuildSynthCdkBuildProjectRoleAA3DC00D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HeseInfraPipelineCodeBuildActionRole5463EFD7": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineRole42FEB153",
                    "Arn",
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "HeseInfraPipelineCodeBuildActionRoleDefaultPolicyE2B3FC80": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "HeseInfraPipelineBuildSynthCdkBuildProjectAD5CA4AD",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "HeseInfraPipelineUpdatePipelineSelfMutationD83C5804",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HeseInfraPipelineCodeBuildActionRoleDefaultPolicyE2B3FC80",
        "Roles": [
          {
            "Ref": "HeseInfraPipelineCodeBuildActionRole5463EFD7",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HeseInfraPipelineRole42FEB153": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "HeseInfraPipelineRoleDefaultPolicy9CD0A2E3": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineArtifactsBucket79D993E5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HeseInfraPipelineArtifactsBucket79D993E5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "HeseInfraPipelineSourcemarkuslheseappCodePipelineActionRole27EDF73B",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "HeseInfraPipelineCodeBuildActionRole5463EFD7",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HeseInfraPipelineRoleDefaultPolicy9CD0A2E3",
        "Roles": [
          {
            "Ref": "HeseInfraPipelineRole42FEB153",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HeseInfraPipelineSourcemarkuslheseappCodePipelineActionRole27EDF73B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "HeseInfraPipelineSourcemarkuslheseappCodePipelineActionRoleDefaultPolicy1B879BE0": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "codestar-connections:UseConnection",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:codestar-connections:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":connection/2176abff-fac4-4c5d-87e8-0cc53551ab98",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineArtifactsBucket79D993E5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HeseInfraPipelineArtifactsBucket79D993E5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:PutObjectAcl",
                "s3:PutObjectVersionAcl",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "HeseInfraPipelineArtifactsBucket79D993E5",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HeseInfraPipelineSourcemarkuslheseappCodePipelineActionRoleDefaultPolicy1B879BE0",
        "Roles": [
          {
            "Ref": "HeseInfraPipelineSourcemarkuslheseappCodePipelineActionRole27EDF73B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HeseInfraPipelineUpdatePipelineSelfMutationD83C5804": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "Description": "Pipeline step infra/Pipeline/UpdatePipeline/SelfMutate",
        "EncryptionKey": "alias/aws/s3",
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:7.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "HeseInfraPipelineUpdatePipelineSelfMutationRoleA055CB95",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": "{
  "version": "0.2",
  "phases": {
    "install": {
      "commands": [
        "npm install -g aws-cdk@2"
      ]
    },
    "build": {
      "commands": [
        "cdk -a . deploy infra --require-approval=never --verbose"
      ]
    }
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "HeseInfraPipelineUpdatePipelineSelfMutationRoleA055CB95": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "HeseInfraPipelineUpdatePipelineSelfMutationRoleDefaultPolicy4466AA5D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "HeseInfraPipelineUpdatePipelineSelfMutationD83C5804",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "HeseInfraPipelineUpdatePipelineSelfMutationD83C5804",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "HeseInfraPipelineUpdatePipelineSelfMutationD83C5804",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Condition": {
                "ForAnyValue:StringEquals": {
                  "iam:ResourceTag/aws-cdk:bootstrap-role": [
                    "image-publishing",
                    "file-publishing",
                    "deploy",
                  ],
                },
              },
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:*:iam::",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":role/*",
                  ],
                ],
              },
            },
            {
              "Action": "cloudformation:DescribeStacks",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "s3:ListBucket",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HeseInfraPipelineArtifactsBucket79D993E5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HeseInfraPipelineArtifactsBucket79D993E5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HeseInfraPipelineUpdatePipelineSelfMutationRoleDefaultPolicy4466AA5D",
        "Roles": [
          {
            "Ref": "HeseInfraPipelineUpdatePipelineSelfMutationRoleA055CB95",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HeseStatus76E9EA78": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": "hese-status",
        "CorsConfiguration": {
          "CorsRules": [
            {
              "AllowedMethods": [
                "GET",
              ],
              "AllowedOrigins": [
                "*",
              ],
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "HeseStatusUpdate824FD175": {
      "DependsOn": [
        "HeseStatusUpdateServiceRoleDefaultPolicy17EFE0D5",
        "HeseStatusUpdateServiceRoleBBBE7064",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "[HASH REMOVED].zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "BUCKET_NAME": {
              "Ref": "HeseStatus76E9EA78",
            },
          },
        },
        "Handler": "index.handler",
        "LoggingConfig": {
          "LogFormat": "JSON",
        },
        "MemorySize": 256,
        "Role": {
          "Fn::GetAtt": [
            "HeseStatusUpdateServiceRoleBBBE7064",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "HeseStatusUpdateServiceRoleBBBE7064": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "HeseStatusUpdateServiceRoleDefaultPolicy17EFE0D5": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HeseStatus76E9EA78",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HeseStatus76E9EA78",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:PutObjectAcl",
                "s3:PutObjectVersionAcl",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "HeseStatus76E9EA78",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HeseStatusUpdateServiceRoleDefaultPolicy17EFE0D5",
        "Roles": [
          {
            "Ref": "HeseStatusUpdateServiceRoleBBBE7064",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HttpApiDefaultStage3EEB07D6": {
      "Properties": {
        "ApiId": {
          "Ref": "HttpApiF5A9A8A7",
        },
        "AutoDeploy": true,
        "StageName": "$default",
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
    "HttpApiF5A9A8A7": {
      "Properties": {
        "Name": "HttpApi",
        "ProtocolType": "HTTP",
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "HttpApiGETstatus15B35F18": {
      "Properties": {
        "ApiId": {
          "Ref": "HttpApiF5A9A8A7",
        },
        "AuthorizationType": "NONE",
        "RouteKey": "GET /status",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "HttpApiGETstatusintegration1E1875D2",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "HttpApiGETstatusintegration1E1875D2": {
      "Properties": {
        "ApiId": {
          "Ref": "HttpApiF5A9A8A7",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": {
          "Fn::GetAtt": [
            "HeseStatusUpdate824FD175",
            "Arn",
          ],
        },
        "PayloadFormatVersion": "2.0",
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "HttpApiGETstatusintegrationPermissionF7424F40": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HeseStatusUpdate824FD175",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "HttpApiF5A9A8A7",
              },
              "/*/*/status",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "Rule4C995B7F": {
      "Properties": {
        "ScheduleExpression": "cron(0/15 * * * ? *)",
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::GetAtt": [
                "HeseStatusUpdate824FD175",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "RuleAllowEventRuleinfraHeseStatusUpdate76B553567F75FDCA": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HeseStatusUpdate824FD175",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": {
          "Fn::GetAtt": [
            "Rule4C995B7F",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "hesereactappC861EE5E": {
      "Properties": {
        "BasicAuthConfig": {
          "EnableBasicAuth": false,
        },
        "IAMServiceRole": {
          "Fn::GetAtt": [
            "hesereactappRoleC1FB01B7",
            "Arn",
          ],
        },
        "Name": "hese-react-app",
        "OauthToken": {
          "Fn::Join": [
            "",
            [
              "{{resolve:secretsmanager:arn:aws:secretsmanager:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":secret:GITHUB_OAUTH_TOKEN-Qw9kGt:SecretString:::}}",
            ],
          ],
        },
        "Platform": "WEB",
        "Repository": "https://github.com/markusl/hese-app",
      },
      "Type": "AWS::Amplify::App",
    },
    "hesereactappRoleC1FB01B7": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "amplify.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "hesereactappmaster53520226": {
      "Properties": {
        "AppId": {
          "Fn::GetAtt": [
            "hesereactappC861EE5E",
            "AppId",
          ],
        },
        "BranchName": "master",
        "EnableAutoBuild": true,
        "EnablePullRequestPreview": true,
      },
      "Type": "AWS::Amplify::Branch",
    },
    "hesereactappolmibeCBE732A4": {
      "Properties": {
        "AppId": {
          "Fn::GetAtt": [
            "hesereactappC861EE5E",
            "AppId",
          ],
        },
        "AutoSubDomainCreationPatterns": [
          "*",
          "pr*",
        ],
        "AutoSubDomainIAMRole": {
          "Fn::GetAtt": [
            "hesereactappRoleC1FB01B7",
            "Arn",
          ],
        },
        "DomainName": "olmi.be",
        "EnableAutoSubDomain": false,
        "SubDomainSettings": [
          {
            "BranchName": {
              "Fn::GetAtt": [
                "hesereactappmaster53520226",
                "BranchName",
              ],
            },
            "Prefix": "hese",
          },
        ],
      },
      "Type": "AWS::Amplify::Domain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
